<launch>
  <node name="push_learning_perception" pkg="tabletop_pushing" type="tabletop_pushing_perception_node" respawn="false" output="screen">
    <remap from="color_image_topic" to="/kinect_head/rgb/image_color"/>
    <remap from="depth_image_topic" to="/kinect_head/depth/image"/>
    <remap from="mask_image_topic" to="/kinect_head/rgb/image_color/self_mask"/>
    <remap from="point_cloud_topic" to="/kinect_head/depth/points"/>

    <param name="use_displays" value="true" type="bool"/>
    <param name="display_wait_ms" value="3" type="int"/>
    <param name="start_tracking_on_push_call" value="false" type="bool"/>
    <param name="write_to_disk" value="true" type="bool"/>
    <param name="write_input_to_disk" value="false" type="bool"/>
    <param name="major_axis_spin_pos_scale" value="0.5" type="double"/>

    <param name="workspace_frame" value="torso_lift_link" type="string"/>
    <param name="min_workspace_x" value="-1.0" type="double"/>
    <param name="max_workspace_x" value="1.75" type="double"/>
    <param name="min_workspace_z" value="-0.8" type="double"/>
    <param name="max_workspace_z" value="0.6" type="double"/>
    <param name="min_table_z" value="-1.0" type="double"/>
    <param name="max_table_z" value="1.0" type="double"/>

    <param name="num_downsamples" value="1" type="int"/>

    <param name="pcl_cluster_tolerance" value="0.03" type="double"/>
    <param name="pcl_min_cluster_size" value="200" type="int"/>
    <param name="pcl_max_cluster_size" value="25000" type="int"/>
    <param name="pcl_voxel_downsample_res" value="0.005" type="double"/>
    <param name="pcl_cloud_intersect_thresh" value="0.0025" type="double"/>
    <param name="pcl_concave_hull_alpha" value="0.1" type="double"/>
    <param name="use_pcl_voxel_downsample" value="false" type="bool"/>
    <param name="table_ransac_thresh" value="0.015" type="double"/>
    <param name="table_ransac_angle_thresh" value="5.0" type="double"/>
    <param name="pcl_difference_thresh" value="0.0001" type="double"/>

    <param name="img_output_path" value="/u/thermans/data/new/" type="string"/>

    <param name="push_tracker_dist_thresh" value="0.05" type="double"/>
    <param name="push_tracker_angle_thresh" value="0.3" type="double"/>
  </node>
</launch>
